stdlib/test/collections/test_optional.mojo

```diff
@@ -52,6 +52,10 @@ def test_basic():
 
     assert_equal(1, (a^).take())
 
+    var a2 = Optional(1)
```

Comment:
**Thought** This is great for testing the non-const reference case.  We don't have a way to test that we return a "const reference to T" in `Optional.value()` right now since we got rid of `let` recently.  Do you mind dropping a `TODO` potentially to come back to this to add a "const reference" test when we have the language facilities to do so?